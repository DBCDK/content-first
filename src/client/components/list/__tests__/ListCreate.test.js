import React from 'react';
import {ListCreator} from '../ListCreate.container';
import renderer from 'react-test-renderer';

jest.mock('react-textarea-autosize', () => 'textarea');
jest.mock('../../general/BookCover.component', () => 'BookCover');
jest.mock('../../general/Link.component', () => 'Link');
jest.mock('../BookSearchSuggester', () => 'BookSearchSuggester');

const profile = {
  name: 'LÃ¦seLotte',
  image: 'http://p-hold.com/200/200'
};

const createTestElement = id => {
  return {
    book: {
      pid: 'pid' + id,
      title: 'some title' + id,
      creator: 'some creator' + id
    },
    links: {
      cover: '/cover' + id
    },
    description: 'some description' + id
  };
};

describe('ListCreate', () => {
  test('list details is shown', () => {
    const currentList = {
      _id: 'current-list-id',
      title: 'some title',
      description: 'some description',
      public: true,
      open: true,
      social: true,
      list: []
    };

    const tree = renderer
      .create(
        <ListCreator
          id="current-list-id"
          profile={profile}
          currentList={currentList}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a list with one book', () => {
    const currentList = {
      _id: 'current-list-id',
      title: 'some title',
      description: 'some description',
      public: false,
      open: false,
      social: false,
      owner: 1,
      list: [createTestElement(1)]
    };
    const tree = renderer
      .create(
        <ListCreator
          id="current-list-id"
          profile={profile}
          currentList={currentList}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a list with multiple books', () => {
    const currentList = {
      _id: 'current-list-id',
      title: 'some title',
      description: 'some description',
      public: false,
      list: [createTestElement(3), createTestElement(2), createTestElement(1)]
    };
    const tree = renderer
      .create(
        <ListCreator
          id="current-list-id"
          profile={profile}
          currentList={currentList}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
