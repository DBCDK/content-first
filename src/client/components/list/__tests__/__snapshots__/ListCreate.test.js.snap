// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCreate CreateList component is rendered 1`] = `
<div
  className="list-creator"
>
  <h1
    className="list-creator__headline"
  >
    Opret liste
  </h1>
  <form
    className="list-creator__form"
    onSubmit={[Function]}
  >
    <div
      className="list-details"
    >
      <div
        className="form-group"
      >
        <span
          className="required "
        >
          <input
            className="form-control"
            name="list-title"
            onChange={[Function]}
            placeholder="Giv din liste en titel"
            type="text"
            value=""
          />
          
        </span>
        <textarea
          className="form-control list-details__description"
          name="list-description"
          onChange={[Function]}
          placeholder="Skriv lidt om din liste"
          value=""
        />
      </div>
    </div>
    <h2
      className="list-creator__headline"
    >
      Tilføj bøger til listen
    </h2>
    <div
      className="list-drag"
    >
      <div
        className="suggestion-list"
      >
        <div
          className="react-autosuggest__container"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup={false}
            aria-owns="react-autowhatever-1"
            autoComplete="off"
            className="form-control suggestion-list__search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Skriv titlen på en bog"
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
          />
        </div>
      </div>
      <div
        className="drag-list "
      />
    </div>
    <div
      className="list-creator__publication"
    >
      <label
        htmlFor="public"
      >
        <input
          checked={false}
          id="public"
          name="public"
          onClick={[Function]}
          type="checkbox"
        />
        <span />
         Skal listen være offentlig?
      </label>
    </div>
    <div
      className="list-creator__submit text-right"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Gem liste
      </button>
    </div>
  </form>
</div>
`;

exports[`ListCreate a book is added to the list 1`] = `
<div
  className="list-creator"
>
  <h1
    className="list-creator__headline"
  >
    Opret liste
  </h1>
  <form
    className="list-creator__form"
    onSubmit={[Function]}
  >
    <div
      className="list-details"
    >
      <div
        className="form-group"
      >
        <span
          className="required "
        >
          <input
            className="form-control"
            name="list-title"
            onChange={[Function]}
            placeholder="Giv din liste en titel"
            type="text"
            value=""
          />
          
        </span>
        <textarea
          className="form-control list-details__description"
          name="list-description"
          onChange={[Function]}
          placeholder="Skriv lidt om din liste"
          value=""
        />
      </div>
    </div>
    <h2
      className="list-creator__headline"
    >
      Tilføj bøger til listen
    </h2>
    <div
      className="list-drag"
    >
      <div
        className="suggestion-list"
      >
        <div
          className="react-autosuggest__container"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup={false}
            aria-owns="react-autowhatever-1"
            autoComplete="off"
            className="form-control suggestion-list__search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Skriv titlen på en bog"
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
          />
        </div>
      </div>
      <div
        className="drag-list "
      >
        <div
          className="drag-item flex"
          onMouseEnter={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <img
            alt="drag book"
            className="drag-item__dragger"
            onMouseDown={[Function]}
            src="drag.svg"
          />
          <div
            className="flex list-item"
          >
            <img
              alt="some title1"
              className="list-item__img"
              src="https://content-first.demo.dbc.dk/cover1"
            />
            <div
              className="list-item__details"
            >
              <div
                className="list-item__title"
              >
                some title1
              </div>
              <div
                className="list-item__author"
              >
                some creator1
              </div>
              <textarea
                className="form-control"
                name="list-description"
                onChange={[Function]}
                placeholder="Skriv en kommentar til bogen"
                value="some description1"
              />
            </div>
          </div>
          <div>
            <span
              className="list-item__close"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="list-creator__publication"
    >
      <label
        htmlFor="public"
      >
        <input
          checked={false}
          id="public"
          name="public"
          onClick={[Function]}
          type="checkbox"
        />
        <span />
         Skal listen være offentlig?
      </label>
    </div>
    <div
      className="list-creator__submit text-right"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Gem liste
      </button>
    </div>
  </form>
</div>
`;

exports[`ListCreate a books is removed from the list 1`] = `
<div
  className="list-creator"
>
  <h1
    className="list-creator__headline"
  >
    Opret liste
  </h1>
  <form
    className="list-creator__form"
    onSubmit={[Function]}
  >
    <div
      className="list-details"
    >
      <div
        className="form-group"
      >
        <span
          className="required "
        >
          <input
            className="form-control"
            name="list-title"
            onChange={[Function]}
            placeholder="Giv din liste en titel"
            type="text"
            value=""
          />
          
        </span>
        <textarea
          className="form-control list-details__description"
          name="list-description"
          onChange={[Function]}
          placeholder="Skriv lidt om din liste"
          value=""
        />
      </div>
    </div>
    <h2
      className="list-creator__headline"
    >
      Tilføj bøger til listen
    </h2>
    <div
      className="list-drag"
    >
      <div
        className="suggestion-list"
      >
        <div
          className="react-autosuggest__container"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup={false}
            aria-owns="react-autowhatever-1"
            autoComplete="off"
            className="form-control suggestion-list__search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Skriv titlen på en bog"
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
          />
        </div>
      </div>
      <div
        className="drag-list "
      >
        <div
          className="drag-item flex"
          onMouseEnter={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <img
            alt="drag book"
            className="drag-item__dragger"
            onMouseDown={[Function]}
            src="drag.svg"
          />
          <div
            className="flex list-item"
          >
            <img
              alt="some title1"
              className="list-item__img"
              src="https://content-first.demo.dbc.dk/cover1"
            />
            <div
              className="list-item__details"
            >
              <div
                className="list-item__title"
              >
                some title1
              </div>
              <div
                className="list-item__author"
              >
                some creator1
              </div>
              <textarea
                className="form-control"
                name="list-description"
                onChange={[Function]}
                placeholder="Skriv en kommentar til bogen"
                value="some description1"
              />
            </div>
          </div>
          <div>
            <span
              className="list-item__close"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="list-creator__publication"
    >
      <label
        htmlFor="public"
      >
        <input
          checked={false}
          id="public"
          name="public"
          onClick={[Function]}
          type="checkbox"
        />
        <span />
         Skal listen være offentlig?
      </label>
    </div>
    <div
      className="list-creator__submit text-right"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Gem liste
      </button>
    </div>
  </form>
</div>
`;

exports[`ListCreate list details is shown 1`] = `
<div
  className="list-creator"
>
  <h1
    className="list-creator__headline"
  >
    Opret liste
  </h1>
  <form
    className="list-creator__form"
    onSubmit={[Function]}
  >
    <div
      className="list-details"
    >
      <div
        className="form-group"
      >
        <span
          className="required "
        >
          <input
            className="form-control"
            name="list-title"
            onChange={[Function]}
            placeholder="Giv din liste en titel"
            type="text"
            value="some title"
          />
          
        </span>
        <textarea
          className="form-control list-details__description"
          name="list-description"
          onChange={[Function]}
          placeholder="Skriv lidt om din liste"
          value="some description"
        />
      </div>
    </div>
    <h2
      className="list-creator__headline"
    >
      Tilføj bøger til listen
    </h2>
    <div
      className="list-drag"
    >
      <div
        className="suggestion-list"
      >
        <div
          className="react-autosuggest__container"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup={false}
            aria-owns="react-autowhatever-1"
            autoComplete="off"
            className="form-control suggestion-list__search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Skriv titlen på en bog"
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
          />
        </div>
      </div>
      <div
        className="drag-list "
      />
    </div>
    <div
      className="list-creator__publication"
    >
      <label
        htmlFor="public"
      >
        <input
          checked={true}
          id="public"
          name="public"
          onClick={[Function]}
          type="checkbox"
        />
        <span />
         Skal listen være offentlig?
      </label>
    </div>
    <div
      className="list-creator__submit text-right"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Gem liste
      </button>
    </div>
  </form>
</div>
`;

exports[`ListCreate multible books are added to the list 1`] = `
<div
  className="list-creator"
>
  <h1
    className="list-creator__headline"
  >
    Opret liste
  </h1>
  <form
    className="list-creator__form"
    onSubmit={[Function]}
  >
    <div
      className="list-details"
    >
      <div
        className="form-group"
      >
        <span
          className="required "
        >
          <input
            className="form-control"
            name="list-title"
            onChange={[Function]}
            placeholder="Giv din liste en titel"
            type="text"
            value=""
          />
          
        </span>
        <textarea
          className="form-control list-details__description"
          name="list-description"
          onChange={[Function]}
          placeholder="Skriv lidt om din liste"
          value=""
        />
      </div>
    </div>
    <h2
      className="list-creator__headline"
    >
      Tilføj bøger til listen
    </h2>
    <div
      className="list-drag"
    >
      <div
        className="suggestion-list"
      >
        <div
          className="react-autosuggest__container"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-expanded={false}
            aria-haspopup={false}
            aria-owns="react-autowhatever-1"
            autoComplete="off"
            className="form-control suggestion-list__search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Skriv titlen på en bog"
            role="combobox"
            type="text"
            value=""
          />
          <div
            className="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
          />
        </div>
      </div>
      <div
        className="drag-list "
      >
        <div
          className="drag-item flex"
          onMouseEnter={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <img
            alt="drag book"
            className="drag-item__dragger"
            onMouseDown={[Function]}
            src="drag.svg"
          />
          <div
            className="flex list-item"
          >
            <img
              alt="some title3"
              className="list-item__img"
              src="https://content-first.demo.dbc.dk/cover3"
            />
            <div
              className="list-item__details"
            >
              <div
                className="list-item__title"
              >
                some title3
              </div>
              <div
                className="list-item__author"
              >
                some creator3
              </div>
              <textarea
                className="form-control"
                name="list-description"
                onChange={[Function]}
                placeholder="Skriv en kommentar til bogen"
                value="some description3"
              />
            </div>
          </div>
          <div>
            <span
              className="list-item__close"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="drag-item flex"
          onMouseEnter={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <img
            alt="drag book"
            className="drag-item__dragger"
            onMouseDown={[Function]}
            src="drag.svg"
          />
          <div
            className="flex list-item"
          >
            <img
              alt="some title2"
              className="list-item__img"
              src="https://content-first.demo.dbc.dk/cover2"
            />
            <div
              className="list-item__details"
            >
              <div
                className="list-item__title"
              >
                some title2
              </div>
              <div
                className="list-item__author"
              >
                some creator2
              </div>
              <textarea
                className="form-control"
                name="list-description"
                onChange={[Function]}
                placeholder="Skriv en kommentar til bogen"
                value="some description2"
              />
            </div>
          </div>
          <div>
            <span
              className="list-item__close"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="drag-item flex"
          onMouseEnter={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <img
            alt="drag book"
            className="drag-item__dragger"
            onMouseDown={[Function]}
            src="drag.svg"
          />
          <div
            className="flex list-item"
          >
            <img
              alt="some title1"
              className="list-item__img"
              src="https://content-first.demo.dbc.dk/cover1"
            />
            <div
              className="list-item__details"
            >
              <div
                className="list-item__title"
              >
                some title1
              </div>
              <div
                className="list-item__author"
              >
                some creator1
              </div>
              <textarea
                className="form-control"
                name="list-description"
                onChange={[Function]}
                placeholder="Skriv en kommentar til bogen"
                value="some description1"
              />
            </div>
          </div>
          <div>
            <span
              className="list-item__close"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="list-creator__publication"
    >
      <label
        htmlFor="public"
      >
        <input
          checked={false}
          id="public"
          name="public"
          onClick={[Function]}
          type="checkbox"
        />
        <span />
         Skal listen være offentlig?
      </label>
    </div>
    <div
      className="list-creator__submit text-right"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Gem liste
      </button>
    </div>
  </form>
</div>
`;
